Assets/scripts/Weapons/Weapon.cs

```diff
@@ -0,0 +1,265 @@
+		[HideInInspector]
+		public double FireCountDown;
+		/// <summary>
+		/// If the weapon is currently in automatic action
+		/// </summary>
+		[HideInInspector]
+		public bool InAutomaticAction;
+
+		[SyncVar(hook="LoadUnloadAmmo")]
+		public NetworkInstanceId MagNetID;
+
+		[SyncVar]
+		public string ControlledByPlayer;
+
+		void Start()
+		{
+			InAutomaticAction = false;
+
+			//init weapon with missing settings
+			if (AmmoType == null)
+				AmmoType = "12mm";
+			if (Projectile == null)
+				Projectile = Resources.Load("Bullet_12mm") as GameObject;
+		}
+
+		void Update () {
+			if (FireCountDown > 0 )
+			{
+				FireCountDown -= Time.deltaTime;
+				//prevents the next projectile taking miliseconds longer than it should
+				if (FireCountDown < 0) {
+					FireCountDown = 0;
+				}
+			}
+
+			if (Input.GetKeyDown (KeyCode.E)) {
+				ManualUnload(CurrentMagazine);
+			}
+
+			if(Input.GetMouseButtonUp(0)) {
+				InAutomaticAction = false;
+			}
+
+			if (InAutomaticAction && FireCountDown <= 0) {
+				AttemptToFireWeapon();
+			}
+		}
+
+		#region Weapon Server Init
+		public override void OnStartServer()
+		{
+			GameObject m = GameObject.Instantiate(Resources.Load("Magazine_"+AmmoType) as GameObject, Vector3.zero, Quaternion.identity);
+			NetworkServer.Spawn(m);
+			StartCoroutine(SetMagazineOnStart(m));
+			base.OnStartServer();
+		}
+
+		//Gives it a chance for weaponNetworkActions to init
+		IEnumerator SetMagazineOnStart(GameObject magazine){
+			yield return new WaitForEndOfFrame();
+			PlayerManager.LocalPlayerScript.weaponNetworkActions.CmdLoadMagazine(gameObject, magazine);
+		}
+		#endregion
+
+		#region Weapon Firing Mechanism
+		public override void Interact() {
+			//shoot gun interation if its in hand
+			if (gameObject == UIManager.Hands.CurrentSlot.GameObject()) {
+				AttemptToFireWeapon();
+			} 
+			//if the weapon is not in our hands not in hands, pick it up
+			else {
+				PlayerManager.LocalPlayerScript.playerNetworkActions.TryToPickUpObject(gameObject);
+			}
+		}
+
+		void AttemptToFireWeapon()
+		{
+			//ignore if we are hovering over UI
+			if (UnityEngine.EventSystems.EventSystem.current.IsPointerOverGameObject())
+				return;
+
+			//if the hand with the weapon in it is selected
+			if (UIManager.Hands.CurrentSlot == UIManager.Hands.LeftSlot || UIManager.Hands.CurrentSlot == UIManager.Hands.RightSlot) {
+				//if we have no mag/clip loaded play empty sound
+				if (CurrentMagazine == null) {
+					InAutomaticAction = false;
+					PlayEmptySFX();
+					return;
+				} 
+				//if we are out of ammo for this weapon play an empty sound
+				else if (Projectile != null && CurrentMagazine.ammoRemains <= 0 && FireCountDown <= 0) {
+					InAutomaticAction = false;
+					PlayEmptySFX();
+					return;
+				}
+				//if we are out of ammo for this weapon play an empty sound - machine gun fire ending
+				else if (Projectile != null && CurrentMagazine.ammoRemains <= 0 && FireCountDown <= 0 && InAutomaticAction) {
```

Comment:
those if's should be swapped around
so 
`else if (Projectile != null && CurrentMagazine.ammoRemains <= 0 && FireCountDown <= 0 && InAutomaticAction) {`
should come first